libraries/fc/src/network/udp_socket.cpp

```diff
@@ -1,166 +1,105 @@
-    my->_sock.async_send_to( boost::asio::buffer(buffer.get(), length), to_asio_ep(to), 
-                             asio::detail::read_write_handler_with_buffer(completion_promise, buffer) );
-
-    return completion_promise->get_future().get();
-  }
-
-  void udp_socket::open() {
-    my->_sock.open( boost::asio::ip::udp::v4() );
-    my->_sock.non_blocking(true);
-  }
-  void udp_socket::set_receive_buffer_size( size_t s ) {
-    my->_sock.set_option(boost::asio::socket_base::receive_buffer_size(s) );
-  }
-  void udp_socket::bind( const fc::ip::endpoint& e ) {
-    my->_sock.bind( to_asio_ep(e) );
+      // Swallow errors.  Currently only used for GELF logging, so depend on local
+      // log to catch anything that doesn't make it across the network.
+    });
   }
 
-  size_t udp_socket::receive_from( const std::shared_ptr<char>& receive_buffer, size_t receive_buffer_length, fc::ip::endpoint& from )
+  void udp_socket::send_to(const std::shared_ptr<const char>& buffer, size_t length,
+                           boost::asio::ip::udp::endpoint& to)
   {
-    try 
+    try
     {
-      boost::asio::ip::udp::endpoint boost_from_endpoint;
-      size_t bytes_read = my->_sock.receive_from( boost::asio::buffer(receive_buffer.get(), receive_buffer_length), 
-                                                  boost_from_endpoint );
-      from = to_fc_ep(boost_from_endpoint);
-      return bytes_read;
+      my->_sock->send_to(boost::asio::buffer(buffer.get(), length), to);
     } 
-    catch( const boost::system::system_error& e ) 
+    catch(const boost::system::system_error& e)
     {
-      if( e.code() != boost::asio::error::would_block ) 
+      if(e.code() != boost::asio::error::would_block)
         throw;
     }
 
-    boost::asio::ip::udp::endpoint boost_from_endpoint;
-    auto completion_promise = std::make_shared<promise<size_t>>();
-    my->_sock.async_receive_from( boost::asio::buffer(receive_buffer.get(), receive_buffer_length), 
-                                  boost_from_endpoint,
-                                  asio::detail::read_write_handler_with_buffer(completion_promise, receive_buffer) );
-    size_t bytes_read = completion_promise->get_future().get();
-    from = to_fc_ep(boost_from_endpoint);
-    return bytes_read;
-  }
-
-  size_t udp_socket::receive_from( char* receive_buffer, size_t receive_buffer_length, fc::ip::endpoint& from ) 
-  {
-    try 
-    {
-      boost::asio::ip::udp::endpoint boost_from_endpoint;
-      size_t bytes_read = my->_sock.receive_from( boost::asio::buffer(receive_buffer, receive_buffer_length), 
-                                                  boost_from_endpoint );
-      from = to_fc_ep(boost_from_endpoint);
-      return bytes_read;
-    } 
-    catch( const boost::system::system_error& e ) 
+    my->_sock->async_send_to(boost::asio::buffer(buffer.get(), length), to,
+                             [this](const boost::system::error_code& /*ec*/, std::size_t /*bytes_transferred*/)
```

Comment:
Same problems here as above. `this` captured and not needed. Nothing protects `buffer` from going out of scope.

Reply:
Fixed.